persistence/pre_persistence/block.go

```diff
@@ -0,0 +1,57 @@
+package pre_persistence
+
+import (
```

Comment:
Should we add a TODO to move this into shared? Just add it in if so, otherwise I might need to understand more.